test/api/exclude-members.test.js

```diff
@@ -0,0 +1,70 @@
+const test = require('tape')
+const { promisify: p } = require('util')
+const { Server, replicate, FeedId } = require('../helpers')
+
+test('tribes.excludeMembers',async t => {
+  const kaitiaki = Server()
+  const newPerson = Server()
+
+  const name = id => {
+    if (id === kaitiaki.id) return 'kaitiaki '
+    if (id === newPerson.id) return 'new person'
+  }
+
+  try {
+    const { groupId, groupInitMsg } = await p(kaitiaki.tribes.create)({})
+    t.true(groupId, 'creates group')
+
+    const authorIds = [
+      newPerson.id,
+      FeedId()
+    ]
+
+    await p(kaitiaki.tribes.invite)(
+      groupId, authorIds, {}
+    )
+
+    await p(replicate)({ from: kaitiaki, to: newPerson, live: false, name })
+
+    let exclude = await p(kaitiaki.tribes.excludeMembers)(groupId, authorIds).catch(err=>t.error(err, err.message))
+
+    exclude = await p(kaitiaki.get)({ id: exclude.key, private: true })
+    const expected = {
+      type: 'group/exclude-member',
+      excludes: authorIds,
+
+      tangles: {
+        members: { root: groupInitMsg.key, previous: [groupInitMsg.key] },
+        group: { root: groupInitMsg.key, previous: [...exclude.content.tangles.group.previous] }
+      },
+      recps: [groupId],
+    }
+    t.deepEqual(exclude.content, expected, 'kaitiaki excluded everyone')
+  } catch (err) {
+    t.fail(err)
+  }
+
+  await Promise.all([
+    p(kaitiaki.close)(true),
```

Comment:
```
tribes.excludeMembers
    ✔ creates group
    kaitiaki  ─> new person
      1 [?]
      2 [?]
      3 group/add-member  ─ new person, undefined

/home/me/prj/ssb/ssb-tribes/index.js:108
        if (err) throw err
                 ^
Error [ReadError]: Database is not open
    at maybeError (/home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/levelup@4.4.0/node_modules/levelup/lib/levelup.js:339:32)
    at LevelUP.put (/home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/levelup@4.4.0/node_modules/levelup/lib/levelup.js:211:7)
    at registerAuthor (/home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/ssb-keyring@1.1.0/node_modules/ssb-keyring/models/membership.js:45:8)
    at /home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/ssb-keyring@1.1.0/node_modules/ssb-keyring/models/membership.js:26:41
    at /home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/async-map.js:32:13
    at /home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sources/values.js:21:7
    at next (/home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/async-map.js:27:9)
    at next (/home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sinks/drain.js:22:11)
    at /home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sinks/drain.js:37:15
    at /home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/async-map.js:39:20 {
  [cause]: undefined
}

Node.js v18.16.0
    ✔ kaitiaki excluded everyone

  test count(48) != plan(null)

  Failed tests: There was 1 failure

    ✖ undefined) undefined

  total:     48
  passing:   48
  failing:   1
  2.529 s
```

Reply:
fixed by adding a 500ms wait at the end and a t.end(), weird 4c98a4a

Reply:
we never needed t.end() in promise tests in tribes2, i wonder what's the difference. and what exactly is happening in the 500ms wait

Reply:
that error says that levelDB is trying to do something with a DB which is not open.

```
at registerAuthor (/home/me/prj/ssb/ssb-tribes/node_modules/.pnpm/ssb-keyring@1.1.0/node_modules/ssb-keyring/models/membership.js:45:8)
```
it's trying to call `registerAuthor` in the keyring... but I guess the keyring was closed (unless you waited 500ms)

Reply:
does the keyring open again after 500ms? that's really confusing